### ISSUE TITLE
Our DWARF info makes no sense

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#15537]
I thought I had filed an issue for this already, but I can't find it. There's a big problem with the DWARF we generated for inlined code. Right now what we're describing is a function inlined into itself, but at a different location and a different file. We need to clean up the scopes we use when generating the debug info.

**Comment by vtjnash** [comment]
I think this will be fixed by https://github.com/JuliaLang/julia/pull/14949 once is merged. IIUC, that is blocked on Jeff figuring out https://github.com/JuliaLang/julia/pull/15000.

**Comment by StefanKarpinski** [comment]
I'm a little confused, @vtjnash – #15000 is closed, what needs to be figured out?

**Comment by carnaval** [comment]
#15000 is not really necessary, only nice to have as it would avoid some duplication of information. #1494 is in need of a good cleanup though, will do that today.

**Comment by JeffBezanson** [comment]
@Keno can you confirm whether this is fixed?

**Comment by Keno** [comment]
I'll take a look later and see if there are remaining issues.

**Comment by KristofferC** [comment]
@Keno Did you find anything?

**Comment by KristofferC** [comment]
Bump @Keno, how are the dwarfs doing?

**Comment by StefanKarpinski** [comment]
![dwarves](https://user-images.githubusercontent.com/153596/27560563-855f2d40-5a79-11e7-80a9-31466db26430.jpg)

They seem pretty jolly as a whole.

**Comment by vtjnash** [comment]
Closing, since no response from OP
